mirrord/layer/tests/issue2744.rs

```diff
@@ -0,0 +1,127 @@
+#![feature(assert_matches)]
+#![warn(clippy::indexing_slicing)]
+
+mod common;
+
+use std::{io::Write, net::SocketAddr, ops::Not, path::Path, time::Duration};
+
+pub use common::*;
+use mirrord_protocol::{
+    ClientMessage, ConnectionId, DaemonMessage,
+    outgoing::{
+        DaemonConnect, DaemonRead, LayerClose, LayerConnect, LayerWrite, SocketAddress,
+        tcp::{DaemonTcpOutgoing, LayerTcpOutgoing},
+    },
+};
+use rstest::rstest;
+use tempfile::NamedTempFile;
+
+#[rstest]
+#[tokio::test]
+#[timeout(Duration::from_secs(15))]
+async fn issue_2744_non_blocking_outgoing_tcp(dylib_path: &Path) {
```

Comment:
Small description of the issue as docs? I think this would be nice for every test we add (especially those that are super specific).